kata-webhook/main.go

```diff
@@ -50,7 +50,7 @@ func annotatePodMutator(ctx context.Context, obj metav1.Object) (bool, error) {
 	}
 
 	for i := range pod.Spec.Containers {
-		if pod.Spec.Containers[i].SecurityContext != nil {
+		if pod.Spec.Containers[i].SecurityContext != nil && pod.Spec.Containers[i].SecurityContext.Privileged != nil {
```

Comment:
hi @fidencio , this change looks good. But, in reason of my ignorance on this subject, allow me to ask one question.

If I understood correctly if this webhook is applied to a non-default namespace then the `Privileged` object *might be* nil, as a result it continues to process (eventually the pod is mutated). My doubt is if the lack of `Privileged` really means the container is not privileged or whether it should be checked another object for non-default namespaces.

Reply:
@wainersm,

I think it does and the reason is here: https://pkg.go.dev/k8s.io/api/core/v1#SecurityContext
There it mentions:
```
	// Run container in privileged mode.
	// Processes in privileged containers are essentially equivalent to root on the host.
	// Defaults to false.
	// +optional
	Privileged *bool `json:"privileged,omitempty" protobuf:"varint,2,opt,name=privileged"`
```
From that I understand that unless it's explicitly set, and set to `true`, it'll be false.

Does this answer your question?

Reply:
@fidencio it does answer! Thanks!